src/Microsoft.AspNetCore.Server.Kestrel/IKestrelServerInformation.cs

```diff
@@ -1,12 +1,19 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using Microsoft.AspNetCore.Server.Kestrel.Filter;
 
 namespace Microsoft.AspNetCore.Server.Kestrel
 {
     public interface IKestrelServerInformation
     {
+        TimeSpan ExecutionTimeout { get; set; }
```

Comment:
KeepAlive and HeadersComplete are reasonable, but I don't think we need or want ExecutionTimeout at the server level.
A) The same functionality can be provided in app/middleware code.
B) The timeout is un-enforceable. You can't force an app to abandon a request, you can only close the socket and trip the token. The RequestAborted CancellationToken is at best a hint. There are no thread aborts like v4.
C) There are known scenarios this breaks with long running requests. It's one more setting on the server to hunt down and fiddle with when that happens.